### PR TITLE
 ✨ [#4356] Add required config for Expoints

### LINKED ISSUE #4356: Add integration with Expoints feedback tool
### Thema / Theme

Frontend

### Omschrijving / Description

Similar to GovMetric, we need to add the Expoints feedback tool.

Documentation: https://documentation.expoints.nl/external/Help/Popup

Desired result: A feedback button is shown on the side/below the form (configurable on the Expoints side). This button shows a popup with questions.

We will get a test-account at Expoints to test the workings. It only works (or button appears) when you accept the cookies, for example on Rotterdam:

2024-06-04-049V4PaLPs

### Added value / Toegevoegde waarde

_No response_

### Aanvullende opmerkingen / Additional context

_No response_

### PR DESCRIPTION
**Pull request by stevenbal**
TODO:
- [ ] how to handle i18n -> this doesn't seem to work on rotterdam.nl either, so we might have to exclude the feedback button for non-Dutch for the time being?

Closes #4356 

**Changes**

* Add the required configuration for Expoints feedback tool

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

File: src/openforms/js/compiled-lang/nl.json [hunk 1]
```diff
@@ -3886,6 +3886,12 @@
       "value": "Aantal dagen dat een sessie bewaard blijft. Laat leeg om de waarde van de algemene configuratie te gebruiken."
     }
   ],
+  "fGTdJz": [
+    {
+      "type": 0,
+      "value": "Het afleiden van adres via tekstveld-componenten is verouderd. Gebruik in de plaats de 'AddressNL'-component (in de 'Speciale velden' groep)."
+    }
+  ],
```
Comment (stevenbal) [inline src/openforms/js/compiled-lang/nl.json:3894]: The translation changes seem to be unrelated to this PR, possibly it was skipped for a previous PR?

**Comment by stevenbal** [comment]
@sergei-maertens the loading of the button works (if I use the settings used by gemeente Rotterdam) 
image, but I run into CORS errors if I click on it. I've asked Expoints to whitelist localhost so I can verify that it works properly

File: src/openforms/js/compiled-lang/nl.json (hunk 1 above)
Comment (sergei-maertens) [reply src/openforms/js/compiled-lang/nl.json]: I'll double check - looks related to the formio-builder version :)

Comment (stevenbal) [reply src/openforms/js/compiled-lang/nl.json]: ahh that makes sense, I was working on the addressNL validation as well, I'll see what happens if I reinstall

File: src/openforms/analytics_tools/models.py [hunk 2]
```diff
@@ -254,6 +258,39 @@ class AnalyticsToolsConfiguration(SingletonModel):
         ),
     )
 
+    expoints_organization_name = models.CharField(
+        _("Expoints organization name"),
+        blank=True,
+        max_length=50,
+        help_text=_(
+            "The name of your organization as registered in Expoints. This is construct the URL "
+            "to communicate with Expoints."
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/analytics_tools/models.py:267]: should it be slug-like if it goes in the URL?

File: src/openforms/js/compiled-lang/nl.json (hunk 1 above)
Comment (stevenbal) [reply src/openforms/js/compiled-lang/nl.json]: You're right, I ran it again and now there are no changes, I've removed the commit

File: src/openforms/analytics_tools/templates/analytics_tools/expoints.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{% comment %}
+  Documentation: https://documentation.expoints.nl/external/Help/Popup
+{% endcomment %}
+{% load solo_tags %}
+{% get_solo 'analytics_tools.AnalyticsToolsConfiguration' as analytics_tools_config %}
+{% if analytics_tools_config.is_expoints_enabled %}
+{{analytics_tools_config.expoints_use_test_mode|json_script:"expoints_test_mode"}}
+<script type="text/javascript" id="expoints-analytics">
```
Comment (sergei-maertens) [inline src/openforms/analytics_tools/templates/analytics_tools/expoints.html:8]: inline script needs our csp nonce

File: src/openforms/analytics_tools/templates/analytics_tools/expoints.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    (function() {
+        let scriptConfig = {instanceUrl: 'https://{{ analytics_tools_config.expoints_organization_name }}.expoints.nl'};
```
Comment (sergei-maertens) [inline src/openforms/analytics_tools/templates/analytics_tools/expoints.html:10]:
```suggestion
        let scriptConfig = {instanceUrl: 'https://{{ analytics_tools_config.expoints_organization_name|escapejs }}.expoints.nl'};
```

File: src/openforms/analytics_tools/templates/analytics_tools/expoints.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        let useTestMode = JSON.parse(document.getElementById('expoints_test_mode').innerText);
+        if (useTestMode) scriptConfig.isTest = true;
+
+        var exp = document.createElement('script');
+        exp.type = 'text/javascript';
+        exp.async = true;
+        exp.onload = function() {
+            window.expoints = lightningjs.require('expoints','https://{{ analytics_tools_config.expoints_organization_name }}.expoints.nl/m/Scripts/dist/expoints-external.min.js');
```
Comment (sergei-maertens) [inline src/openforms/analytics_tools/templates/analytics_tools/expoints.html:18]: needs `escapejs`

File: src/openforms/analytics_tools/templates/analytics_tools/expoints.html [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+            expoints('start', '{{ analytics_tools_config.expoints_config_uuid }}', scriptConfig);
```
Comment (sergei-maertens) [inline src/openforms/analytics_tools/templates/analytics_tools/expoints.html:19]: needs `escapejs`

File: src/openforms/analytics_tools/templates/analytics_tools/expoints.html [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        };
+        exp.src = 'https://{{ analytics_tools_config.expoints_organization_name }}.expoints.nl/m/Scripts/dist/expoints-external-loader.min.js';
```
Comment (sergei-maertens) [inline src/openforms/analytics_tools/templates/analytics_tools/expoints.html:21]: needs `escapejs`

File: src/openforms/analytics_tools/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevenbal) [reply src/openforms/analytics_tools/models.py:267]: Yes, should I add that info to the helptext or should I make it a `SlugField`?

File: src/openforms/analytics_tools/models.py [hunk 8]
```diff
@@ -403,4 +448,10 @@ def clean(self):
                     "If you enable GovMetric, you need to provide the source ID for all languages (the same one can be reused)."
                 )
             )
+        if self.enable_expoints_analytics and not self.is_expoints_enabled:
+            raise ValidationError(
+                _(
+                    "If you enable GovMetric, you need to provide the source ID for all languages (the same one can be reused)."
```
[the remark this answers is not in the document]
Comment (stevenbal) [reply src/openforms/analytics_tools/models.py:454]: whoops

File: src/openforms/analytics_tools/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/analytics_tools/models.py:267]: slugfield :)

File: src/openapi.yaml [hunk 9]
```diff
@@ -26,8 +26,9 @@ paths:
   /api/v2/analytics/analytics_tools_config_info:
     get:
       operationId: analytics_analytics_tools_config_info_retrieve
-      description: Haal informatie op over de analytics-tools voor de frontend.
-      summary: Analytics tools-configuratie
+      description: Returns information about the analytics tools that are needed by
+        the frontend.
+      summary: Analytics Tools Configuration
```
Comment (sergei-maertens) [inline src/openapi.yaml:31]: I think you may need to run `compilemessages` locally?

**Review by sergei-maertens — APPROVED** [review]
Nice, we'll be able to include this in 2.7 like this :)
